controllers/admin/AdminSearchController.php

```diff
@@ -98,7 +98,7 @@ public function postProcess()
             if (!$searchType || $searchType == 1) {
                 /* Handle product ID */
                 if ($searchType == 1 && (int) $this->query && Validate::isUnsignedInt((int) $this->query)) {
-                    $product = new Product($this->query);
+                    $product = new Product((int) $this->query);
```

Comment:
Looks a bit confusing an redundant:
```suggestion
                    $productId = (int) $this->query;
                    $product = new Product($productId);
```
This would keep simple the code below.